src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,867 @@
+        require(delegation.isOperator(operator), OperatorNotRegistered());
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            _updateFreeMagnitude({operator: operator, strategy: strategies[i], numToComplete: numToComplete[i]});
+        }
+    }
+
+    /**
+     * @notice Modifies the propotions of slashable stake allocated to a list of operatorSets for a set of strategies
+     * @param operator address to modify allocations for
+     * @param allocations array of magnitude adjustments for multiple strategies and corresponding operator sets
+     * @param operatorSignature signature of the operator if msg.sender is not the operator
+     * @dev Updates freeMagnitude for the updated strategies
+     * @dev Must be called by the operator or with a valid operator signature
+     * @dev For each allocation, allocation.operatorSets MUST be ordered in ascending order according to the
+     * encoding of the operatorSet. This is to prevent duplicate operatorSets being passed in. The easiest way to ensure
+     * ordering is to sort allocated operatorSets by address first, and then sort for each avs by ascending operatorSetIds.
+     */
+    function modifyAllocations(
+        address operator,
+        MagnitudeAllocation[] calldata allocations,
+        SignatureWithSaltAndExpiry calldata operatorSignature
+    ) external {
+        if (msg.sender != operator) {
+            _verifyOperatorSignature(operator, allocations, operatorSignature);
+        }
+        require(delegation.isOperator(operator), OperatorNotRegistered());
+
+        (bool isSet, uint32 delay) = allocationDelay(operator);
+
+        require(isSet, UninitializedAllocationDelay());
+
+        // effect timestamp for allocations to take effect. This is configurable by operators
+        uint32 effectTimestamp = uint32(block.timestamp) + delay;
+        // completable timestamp for deallocations
+        uint32 completableTimestamp = uint32(block.timestamp) + DEALLOCATION_DELAY;
+
+        for (uint256 i = 0; i < allocations.length; ++i) {
+            // 1. For the given (operator,strategy) clear all pending free magnitude for the strategy and update freeMagnitude
+            _updateFreeMagnitude({
+                operator: operator,
+                strategy: allocations[i].strategy,
+                numToComplete: type(uint16).max
+            });
+
+            // 2. Check current totalMagnitude matches expected value. This is to check for slashing race conditions
+            // where an operator gets slashed from an operatorSet and as a result all the configured allocations have larger
+            // proprtional magnitudes relative to eachother. This check prevents any surprising behavior.
+            uint64 currentTotalMagnitude = _getLatestTotalMagnitude(operator, allocations[i].strategy);
+            require(currentTotalMagnitude == allocations[i].expectedTotalMagnitude, InvalidExpectedTotalMagnitude());
```

Comment:
can this check be done before (1)? it's generally better to revert earlier, all else being equal